tests/diagnostics/token-paste-location.slang.expected

```diff
@@ -0,0 +1,7 @@
+result code = -1
+standard error = {
+token paste(1): error 20001: unexpected '%', expected identifier
+tests/diagnostics/token-paste-location.slang(10): note: see token pasted location
```

Comment:
Pasting `%` and `%` is perhaps a bit confusing. The traditional C preprocessor rules are that the result of a token paste must be a single valid token, but in this case we are pasting `%` and `%` to get the text `%%` and then tokenizing them back into `%` and `%` (hence why the error says `unexpected '%'` instead of `unexpected '%%'`).

This behavior in Slang (where we re-tokenize after a paste, and (seemingly) don't give an error when more than one token is produced) could be considered a bug, and users might want us to fix it down the line (even if I'm inclined to keep it since it is internally consistent).

If we want this test case to not care about this semantic wrinkle, it might be best to use another character than `%`; using `+` would seemingly produce similar results without the semantic question.